src/Common/ColumnsHashingImpl.h

```diff
@@ -188,6 +225,12 @@ class HashMethodBase
             else
                 cache.value = Value();
         }
+        if constexpr (nullable)
+        {
+
+            null_map = &checkAndGetColumn<ColumnNullable>(column)->getNullMapColumn();
+            has_null_data = (null_map->getRatioOfDefaultRows() < 1.0);
```

Comment:
let's replace it with a straightforward loop.

Reply:
It doesn't seem to affect performance here, I have deleted the relevant code